csrc/python_frontend/fusion_definition.cpp

```diff
@@ -351,13 +351,18 @@ std::optional<size_t> FusionDefinition::id() const {
 
 Scalar FusionDefinition::defineScalar() {
   FUSER_PERF_SCOPE("FusionDefinition::defineScalar");
+  NVF_CHECK(
+      trie_node_ != nullptr,
+      "defineScalar must be called inside a 'with' block");
   Scalar out(recording_state_.size(), this);
   recording_state_.emplace_back(out(), serde::StateType::Scalar);
   return out;
 }
 
 Tensor FusionDefinition::addTensor(TensorView* tv) {
   FUSER_PERF_SCOPE("FusionDefinition::addTensor");
+  NVF_CHECK(
+      trie_node_ != nullptr, "AddTensor must be called inside a 'with' block");
```

Comment:
```suggestion
      trie_node_ != nullptr, "addTensor() must be called from an initialized definition via a python context manager or a child class' definition() method.");
```

Reply:
I don't think this method is actually exposed via python but it doesn't hurt to add the check.